frontend/public/module/k8s/types.ts

```diff
@@ -374,6 +374,43 @@ export type HPAMetric = {
     };
   };
 };
+
+type CurrentObject = {
+  averageUtilization?: number;
+  averageValue?: string;
+  value?: string;
+};
+
+type MetricObject = {
+  name: string;
+  selector: Selector;
+};
+
+type CurrentMetrics = {
+  external: {
+    current: CurrentObject;
+    metric: MetricObject;
+  };
+  object: {
+    current: CurrentObject;
+    describedObject: {
+      apiVersion?: string;
+      kind: string;
+      name: string;
+    };
+    metric: MetricObject;
+  };
+  pods: {
+    current: CurrentObject;
+    metric: MetricObject;
+  };
+  resource: {
+    name: string;
+    current: CurrentObject;
+  };
```

Comment:
Many of these are optional props.

```suggestion
  external?: {
    current: CurrentObject;
    metric: MetricObject;
  };
  object?: {
    current: CurrentObject;
    describedObject: {
      apiVersion?: string;
      kind: string;
      name: string;
    };
    metric: MetricObject;
  };
  pods?: {
    current: CurrentObject;
    metric: MetricObject;
  };
  resource?: {
    name: string;
    current: CurrentObject;
  };
```